### PR TITLE
feat: update go boilerplate

### PR DESCRIPTION
**Pull request by myugen**
Hi!! It would be nice if we update the go version of this amazing Kata, using Go Modules and following standard's project layout.

This boilerplate has not been updated for three years and is nothing like what is expected of a project. I understand that the _quid_ of this Kata is that it starts from a point that is not correct, but at least it would be nice if it could have some resemblance to a Go project.

### CODE REVIEW

File: go/main.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+			fmt.Println(err.Error())
+			os.Exit(1)
+		}
+		days++
+	}
+
+	for day := 0; day < days; day++ {
+		fmt.Printf("-------- day %d --------\n", day)
+		fmt.Println("Name, SellIn, Quality")
+		for i := 0; i < len(items); i++ {
```
Comment (emilybache) [inline go/main.go:27]: I can't see where "items" comes from. Did you forget to copy them here from texttest_fixture.go?

Comment (emilybache) [reply go/main.go:27]: or probably I just don't understand go well enough.. can you put them in this file anyway?

File: go/gildedrose/gildedrose_test.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package gildedrose_test
+
+import (
+	"REPO_URL/YOUR-USER-OR-ORG-NAME/YOUR-REPO-NAME/gildedrose"
```
Comment (emilybache) [inline go/gildedrose/gildedrose_test.go:4]: do you want to replace REPO_URL/YOUR-USER etc with github.com/emilybache, or is there a good reason not to?

File: go/main.go (hunk 1 above)
Comment (myugen) [reply go/main.go:27]: `items` is a global variable set in the file **texttestfixtures.go**. But I've been thinking that maybe it's better to roll back the changes, leaving the `texttest_fixtures.go` file as it was.

File: go/gildedrose/gildedrose_test.go (hunk 2 above)
Comment (myugen) [reply go/gildedrose/gildedrose_test.go:4]: Yes, I can change it, of course. I opted to indicate it as a template, but there is no other reason honestly.

**Comment by emilybache** [comment]
Generally I'm really pleased to see this update, thankyou! I had a couple of comments, nothing that important really.

**Comment by myugen** [comment]
I have made the changes based on the comments we had.

**Comment by emilybache** [comment]
Thankyou! looks good

**Pull request merged**